apps/src/netsim/NetSimConnection.js

```diff
@@ -94,7 +94,10 @@ var NetSimConnection = module.exports = function (sentLog, receivedLog) {
   this.statusChanges = new ObservableEvent();
 
   // Bind to onBeforeUnload event to attempt graceful disconnect
-  window.addEventListener('beforeunload', this.onBeforeUnload_.bind(this));
+  // Null-guard so constructing this object is test-friendly.
+  if (window && window.addEventListener) {
+    window.addEventListener('beforeunload', this.onBeforeUnload_.bind(this));
+  }
```

Comment:
Is there an equivalent way we can do this in tests? Right now it means there's a divergence between the code that runs in our app, and the code that we test. May not be a big deal in this case, but wold be nice to avoid these divergences if possible.